lib/dry/types/hash/schema.rb

```diff
@@ -4,15 +4,15 @@ class Hash < Definition
       class Schema < Hash
         attr_reader :member_types
 
-        def initialize(primitive, options = {})
+        def initialize(_primitive, options)
           @member_types = options.fetch(:member_types)
           super
         end
 
         def call(hash, meth = :call)
           member_types.each_with_object({}) do |(key, type), result|
             if hash.key?(key)
-              result[key] = type.__send__(meth, hash[key])
+              result[key] = type.public_send(meth, hash.fetch(key))
```

Comment:
> Let's stick to `public_send for now and I'll look into total performance impact at the end, once we're done with refactorings, it may, or may not be an issue. We'll see :)

Great thats all I really care about here. Basically: I'm cool with scrutinizing the refactored code for performance hotspots but this PR is just for me to get the code into a state that I'm confident in refactoring